public/app/features/scenes/components/layout/SceneFlexLayout.tsx

```diff
@@ -0,0 +1,187 @@
+
+export type FlexLayoutDirection = 'row' | 'column' | 'row-reverse' | 'column-reverse';
+export type FlexLayoutJustifyContent =
+  | 'start'
+  | 'end'
+  | 'center'
+  | 'left'
+  | 'right'
+  | 'space-between'
+  | 'space-around'
+  | 'space-evenly'
+  | 'stretch'
+  | 'baseline'
+  | 'first baseline'
+  | 'last baseline'
+  | 'safe center'
+  | 'unsafe center';
+
+export type FlexLayoutAlignContent =
+  | 'start'
+  | 'end'
+  | 'center'
+  | 'space-between'
+  | 'space-around'
+  | 'space-evenly'
+  | 'stretch'
+  | 'baseline'
+  | 'first baseline'
+  | 'last baseline'
+  | 'safe center'
+  | 'unsafe center';
+
+export type FlexLayoutAlignItems =
+  | 'start'
+  | 'end'
+  | 'center'
+  | 'stretch'
+  | 'self-start'
+  | 'self-end'
+  | 'baseline'
+  | 'first baseline'
+  | 'last baseline'
+  | 'safe center'
+  | 'unsafe center';
+
+export type FlexLayoutGap = number | string;
+
+interface SceneFlexLayoutState extends SceneLayoutState {
+  direction?: FlexLayoutDirection;
+  justifyContent?: FlexLayoutJustifyContent;
+  alignContent?: FlexLayoutAlignContent;
+  alignItems?: FlexLayoutAlignItems;
+  gap?: FlexLayoutGap;
+  children: SceneObject[];
+}
+
+interface SceneFlexChildState extends SceneFlexLayoutState {}
+
+export class SceneFlexChild extends SceneObjectBase<SceneFlexChildState> {
+  static Component = FlexLayoutChildComponent;
+
+  getLayout(): SceneFlexLayout {
+    if (this.parent instanceof SceneFlexLayout) {
+      return this.parent;
+    }
+    throw new Error('SceneFlexChild must be a child of SceneFlexLayout');
+  }
+}
+
+export class SceneFlexLayout extends SceneObjectBase<SceneFlexLayoutState> {
+  static Component = FlexLayoutRenderer;
+  static Editor = FlexLayoutEditor;
+
+  toggleDirection() {
+    this.setState({
+      direction: this.state.direction === 'row' ? 'column' : 'row',
+    });
+  }
+}
+
+function renderNodes(nodes: SceneObject[], direction: FlexLayoutDirection, isEditing: boolean): React.ReactNode {
+  return nodes.map((node) => {
+    return <node.Component key={node.state.key} model={node} isEditing={isEditing} />;
+  });
+}
+
+function FlexLayoutRenderer({ model, isEditing }: SceneComponentProps<SceneFlexLayout>) {
+  const {
+    direction = 'row',
+    justifyContent = 'start',
+    alignItems = 'normal',
+    alignContent = 'normal',
+    gap = 8,
+    children,
+  } = model.useState();
+
+  return (
+    <div
+      style={{
+        flexGrow: 1,
+        flexDirection: direction,
+        display: 'flex',
+        gap,
+        justifyContent,
+        alignItems,
+        alignContent,
+        height: '100%',
+      }}
+    >
+      {renderNodes(children, direction, Boolean(isEditing))}
+    </div>
+  );
+}
+
+export function FlexLayoutChildComponent({ model, isEditing }: SceneComponentProps<SceneFlexChild>) {
+  const { children, size } = model.useState();
+  const { direction } = model.getLayout().useState();
```

Comment:
this feels more complex than before, having every child also need to subscribe to the layout/parent state. not convinced we need specific scene objects layout children. 

The type SceneObjectSize can support both options, and we could rename it to be SceneLayoutChildOptions? or we name it gridPos like before and make it support both grid and flex